lib/mixlib/shellout/unix.rb

```diff
@@ -195,14 +182,10 @@ def set_cwd
         Dir.chdir(cwd) if cwd
       end
 
-      # Process group id of the child. Returned as a negative value so you can
-      # put it directly in arguments to kill, wait, etc.
-      #
-      # This may be nil if the child dies before the parent can query the
-      # system for its pgid (on some systems it is an error to get the pgid of
-      # a zombie).
+      # Since we call setsid the child_pgid will be the child_pid, set to negative here
+      # so it can be directly used in arguments to kill, wait, etc.
       def child_pgid
-        @child_pgid
+        -@child_pid
```

Comment:
Yeah I don't think its visible to users and I didn't want to change the existing internal API.